src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let newState = Object.assign({}, state);
+  const caseDefault = 'An error because of an unknown type';
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
You can also make a destructuring to get all the needed props of action (type, extraData, keysToRemove)